plugins/auth/src/pam_password.cpp

```diff
@@ -120,6 +123,51 @@ namespace
 
         THROW(SYS_FORK_ERROR, fmt::format("execl failed [errno:{}]", errno));
     }  // run_pam_auth_check
+
+    auto throw_if_password_exceeds_configured_maximum_length(RsComm& _comm, const std::string& _password) -> void
+    {
+        irods::experimental::scoped_privileged_client spc{_comm};
+
+        GridConfigurationInput input{};
+        std::strncpy(input.name_space, "pam_password", sizeof(input.name_space));
+        std::strncpy(input.option_name, "password_length", sizeof(input.option_name));
+
+        GridConfigurationOutput* output{};
+        irods::at_scope_exit free_output{[&output] { std::free(output); }};
+
+        if (const auto ec = rs_get_grid_configuration_value(&_comm, &input, &output); ec != 0) {
+            char* error_subname = nullptr;
+            const char* error_name = rodsErrorName(ec, &error_subname);
+            const auto msg =
+                fmt::format("Failed to get pam_password max length [ec={}][msg={}]", error_name, error_subname);
+
+            std::free(error_subname);
+
+            THROW(ec, msg);
+        }
+
+        try {
+            const rodsLong_t max_password_length = std::stoll(output->option_value);
+
+            if (max_password_length < 1) {
+                THROW(
+                    CONFIGURATION_ERROR, "Configured max_password_length for pam_password must be a positive integer.");
+            }
+
+            if (static_cast<rodsLong_t>(_password.size()) > max_password_length) {
+                THROW(PASSWORD_EXCEEDS_MAX_SIZE,
+                      fmt::format("PAM password cannot exceed [{}] characters.", max_password_length));
+            }
+        }
+        catch (const std::invalid_argument& e) {
+            THROW(CONFIGURATION_ERROR,
+                  "Configured max_password_length for pam_password cannot be converted to an integer.");
```

Comment:
`max_password_length` ... or just `password_length` ?

Reply:
Yes, need to update the actual config name.